### DISCUSSION TITLE
How to backup ddev project

### DISCUSSION THREAD

**Discussion by christophe-natale** [ddev/ddev#6006] [General]
Hi, I just discover DDEV for drupal which is amazing,
I installed a WSL server on windows.
I use googledrive generally for backup files. But I can't backup DDEV files because of linux WSL server.
I can easilly backup database with snapshot, but I wonder how to automatically backup project folder files ?
Thanks,

**Comment by stasadev** [comment]
Hi @christophe-natale,

> how to automatically backup project folder files ?

I think the most common way to have your project with its history is to use `git`.

> I use googledrive generally for backup files. But I can't backup DDEV files because of linux WSL server.

One way is to use `tar` or `zip` to backup the project to `/mnt/c/...` and then use Google Drive.

**Reply by rfay** [reply]
Backing up is a super-important topic in WSL2 and everywhere else.

As @stasadev says, the first line of defense is pushing your code up with git. That's the fundamental thing I do.

Also, do regular `ddev snapshot -a`

But you probably want a full backup system, so [google is your friend](https://www.google.com/search?q=backing+up+wsl2+files&oq=backing+up+wsl2+files&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIHCAEQIRigATIHCAIQIRigAdIBCDQ4NTVqMGo0qAIAsAIA&sourceid=chrome&ie=UTF-8)

**Comment by christophe-natale** [comment]
Thanks @stasadev and @rfay,
My vscode is now ready for git, first commit create !

**Reply by rfay** [reply]
Here's an article that might be interesting to you. It's from years ago, but still useful: https://ddev.com//blog/save-as-you-go-and-make-backups-in-a-jiffy-with-ddev-local/

**Reply by christophe-natale** [reply]
Thank's Randy, very usefull. DDEV and vscode are powerfull, and propose a unique solution for manage db, files and code. But for backup, we still need to save files on one side and save db on the other side. I'm just trying to deal with git and same thing, there no code for extracting db and make a full archive with files and db.
